### ISSUE TITLE
Context menu not showing up

### ISSUE THREAD

**Issue by oijqwojviqwqp** [spesmilo/electrum#8572]
Right-click context menu does not open for me. Tried in 4.3.1 and 4.4.5 AppImage. More specifically:

- it does not show at all on tabs History, Address, Coins
- in Send tab it shows for few milliseconds when I I right-click in Pay to and then dissapears
- In other text fields (I guess without specific items) it shows fine (e.g. in Description, Amount on the Pay to tab)

I can double-click all the items and the detailed window shows correctly.

Since it worked before for me in 4.3.1 I assume it is related to some Ubuntu update. I am running Ubuntu 23.04, updated.

$XDG_SESSION_TYPE - wayland
$XDG_CURRENT_DESKTOP - GNOME

**Comment by sha-265** [comment]
Confirmed. When using `XDG_SESSION_TYPE=wayland` there is no issue, but the window theme is wrong.

**Comment by mariaa144** [comment]
`XDG_SESSION_TYPE=wayland` doesn't help me on NixOS. Is there are workaround if I need the context menu?

**Comment by SomberNight** [comment]
> Right-click context menu does not open for me
> I assume it is related to some Ubuntu update. I am running Ubuntu 23.04, updated.
> `$XDG_SESSION_TYPE - wayland`
> `$XDG_CURRENT_DESKTOP - GNOME`

> Confirmed. When using `XDG_SESSION_TYPE=wayland` there is no issue

These statements conflict.
@oijqwojviqwqp said they are seeing the issue with `XDG_SESSION_TYPE=wayland`.
@sha-265 is saying setting that fixes it. What exactly is "confirmed"?

---

> `XDG_SESSION_TYPE=wayland` doesn't help me on NixOS. Is there are workaround if I need the context menu?

What about `XDG_SESSION_TYPE=x11`?

**Comment by SomberNight** [comment]
> When using `XDG_SESSION_TYPE=wayland` [...] the window theme is wrong.

Yes, indeed, there are some issues with the theme. (testing on ubuntu 23.04)

Most notably, the titlebar of `QMessageBox` is missing, and so our popups cannot be moved. Also they are positioned unintuitively.
Note: these windows being `WindowModal` seems to be causing this:
https://github.com/spesmilo/electrum/blob/4978d4095673c22030fda6f7f134a0704de42604/electrum/gui/qt/util.py#L297

<details>
 <summary>see picture</summary>

![pic2](https://github.com/spesmilo/electrum/assets/29142493/4c76ba36-74ab-413f-b28c-a2cfce22265b)
</details>

This happens with both the 4.4.6 appimage, and when running from source (both when installing pyqt5 from pip or from apt).

---

Note: same theme weirdness and `WindowModal` issues on ubuntu 22.04 -- but there the apt-provided python3-pyqt5 is missing the wayland plugin, so to even test one would want to install pyqt5 from pip. (`pyqt5==5.15.9`)

---

To clarify/reiterate, I could not reproduce the original issue of "context menu not showing up"; and everything seems to be working for me with `XDG_SESSION_TYPE=x11 QT_QPA_PLATFORM=xcb`, which is essentially the behaviour by default.

(by default I am seeing `XDG_SESSION_TYPE=wayland QT_QPA_PLATFORM=""` but that gets converted to `XDG_SESSION_TYPE=x11 QT_QPA_PLATFORM=xcb` by Qt, when the following line is logged: 
`Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.`
)

**Comment by sha-265** [comment]
> @sha-265 is saying setting that fixes it. What exactly is "confirmed"?

I confirmed that the context menu not showing up, when **not** using `XDG_SESSION_TYPE=wayland`

**Comment by mariaa144** [comment]
For me the menus show up inconsistently. About 10% of the time when I right click the menu WILL show up. The rest of the time nothing happens and I sit there wondering if I'm crazy.

**Comment by SomberNight** [comment]
Ok, so sounds like the context menu missing issue is specific to x11.
Any clues how to reproduce? Like I said, works for me on ubuntu.
@sha-265 what distro are you using?

**Comment by sha-265** [comment]
I'm using Ubuntu 23.04. It's happening both running from source, and installed with `pip`.

**Comment by oijqwojviqwqp** [comment]
Out of pure desperation, I did clean re-install of Ubuntu 23.04. By clean I mean disk format and install from USB.

It worked (!) until yesterday and today it is not working again. There is not anything new I would install except of the Ubuntu update packages.

Are there any additional logs / debugging info I can provide? This drives me crazy.

**Comment by Reza-qezell** [comment]
Hello fellas. Can u please charge my wallet? Have no laptop to contribute more.

0xC9fAfC96958A696F1363D7933f63175435D25B12



Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: oijqwojviqwqp ***@***.***>
Sent: Thursday, August 10, 2023 11:53:55 PM
To: spesmilo/electrum ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [spesmilo/electrum] Context menu not showing up (Issue #8572)


Right-click context menu does not open for me. Tried in 4.3.1 and 4.4.5 AppImage. More specifically:

  *   it does not show at all on tabs History, Address, Coins
  *   in Send tab it shows for few milliseconds when I I right-click in Pay to and then dissapears
  *   In other text fields (I guess without specific items) it shows fine (e.g. in Description, Amount on the Pay to tab)

I can double-click all the items and the detailed window shows correctly.

Since it worked before for me in 4.3.1 I assume it is related to some Ubuntu update. I am running Ubuntu 23.04, updated.

$XDG_SESSION_TYPE - wayland
$XDG_CURRENT_DESKTOP - GNOME
